typo3-fractor/rules/TYPO3v12/FlexForm/MigrateEmailFlagToEmailTypeFlexFormFractor.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+namespace a9f\Typo3Fractor\TYPO3v12\FlexForm;
+
+use a9f\Fractor\Helper\ArrayUtility;
+use a9f\Fractor\Helper\StringUtility;
+use a9f\FractorXml\Helper\FlexFormHelperTrait;
+use a9f\Typo3Fractor\AbstractFlexformFractor;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.0/Feature-97013-NewTCATypeEmail.html
+ * @see \a9f\Typo3Fractor\Tests\Rules\TYPO3v12\FlexForm\MigrateEmailFlagToEmailTypeFlexFormFractor\MigrateEmailFlagToEmailTypeFlexFormFractorTest
+ */
+final class MigrateEmailFlagToEmailTypeFlexFormFractor extends AbstractFlexformFractor
+{
+    use FlexFormHelperTrait;
+
+    private \DOMDocument $domDocument;
+
+    public function canHandle(\DOMNode $node): bool
+    {
+        return parent::canHandle($node) && $node->nodeName === 'config';
+    }
+
+    public function beforeTraversal(\DOMDocument $rootNode): void
+    {
+        $this->domDocument = $rootNode;
```

Comment:
this is kinda ugly but I don't know if there is a better way to pass the DomDocument around

Reply:
We need some kind of context here, that's true. This should be centrally maintained in the DOMDocumentIterator and passed on to each single Fractor.

Reply:
While we don't have that, using DOMNode::parentElement and maintaining this local variable is probably enough. So let's merge this PR as is for now, but keep this in mind for later usage.